python_tests/pytest_input_generators.py

```diff
@@ -1478,3 +1478,25 @@ def vector_at_error_generator(
         yield SampleInput(
             make_arg(error_case["tensor_shape"]), index=error_case["index"]
         ), error_type, error_msg
+
+
+def matmul_input_generator(
+    op: OpInfo, dtype: torch.dtype, requires_grad: bool = False, **kwargs
+):
+    make_arg = partial(
+        make_tensor,
+        dtype=torch.float16,
+        device="cuda",
+        low=None,
+        high=None,
+        requires_grad=requires_grad,
+    )
+    test_cases = (
```

Comment:
Just a suggestion. Feel free to leave it for a future PR. A cartesian product would give better coverage of matrix shapes without much manual effort.

```python
import itertools
import math

def multiply_range(maximum, step):
    assert maximum % step == 0
    num_steps = int(math.log(maximum, step))
    return tuple(map(pow, itertools.repeat(step, num_steps), range(1, num_steps + 1)))

# Ranges of tensor sizes: 8, 64, 512, 4096, 32768
# Use a Cartesian product to create a wide range of matrix shapes
M, N, K = itertools.repeat(multiply_range(32768, 8), 3)
for M, N, K in itertools.product(M, N, K):
    lhs_shape = (M, K)
    rhs_shape = (K, N)
    yield SampleInput(lhs, rhs)

Reply:
Thanks, incorporated